src/pages/docs/concepts-reliability.mdx

```diff
@@ -0,0 +1,363 @@
+that ensure data integrity, maintain consistent performance, and provide a
+seamless experience for end-users, even in the face of various challenges and
+failure scenarios.
+
+# High-Reliability with Golem
+
+With any programming language, tech stack, or SDK, Golem automatically provides
+you with a high degree of built-in reliability.
+
+These guarantees are summarized as follows:
+
+- **Transactional execution**. Golem ensures that your workers execute
+  transactionally. Once they begin execution, they will complete execution,
+  even if there are software or hardware faults, restarts, or updates.
+- **Durable state**. Golem ensures that the state of your workers is durable.
+  If a worker is recovered after a failure, it will resume with the same state
+  as before the failure.
+- **Reliable internal communication**. Golem ensures that communication
+  between workers is delivered reliably. If a message is sent, it will
+  eventually be delivered, with exactly-once semantics; even if there are
+  network partitions, faults, or other failures.
+- **Resilient external communication**. Golem automatically applies retry
+  strategies and other techniques to ensure that transient failures of
+  external systems do not affect reliability.
+
+In the following sections, we will explore these guarantees in more detail,
+looking at how each of them improves different faucets of reliability.
+
+## Resiliency
+
+As discussed in [Components](concepts-components.mdx), all components deployed
+on Golem ultimately derive their capability to interact with external systems
+from _WASI_, which is the Posix-like interface provided by the WASM component
+model.
+
+WASI provides facilities to interact with file systems, sockets, time, and other
+system resources in a portable and secure manner.
+
+Because Golem provides a custom implementation of WASI, it has direct insight
+into the interactions between components and the underlying system. This allows
+Golem to detect a variety of transient failure scenarios and automatically
+apply resiliency strategies, such as retries, to ensure that the system remains
+resilient to these failures.
+
+Golem automatically applies retry strategies to the following types of failures:
+
+- HTTP Request Failures
+
+In the case of errors which are not resolved on their own, Golem will
+eventually give up and mark the worker as failed. This is to prevent the
+system from getting stuck in an infinite loop of retries.
+
+These workers can be detected using worker enumeration.
+
+### Limitations
+
+Golem does not have awareness into transient errors that are expressed in protocol-
+agnostic ways. For example, if an HTTP response contains a 200 status code, but
+the body contains a transient error, then Golem will not be able to detect this
+error and attempt recovery.
+
+You should be aware of this limitation when designing your applications, and
+consider implementing your own retry strategies for such cases.
+
+### Customization
```

Comment:
This is not completely true, you can already change the worker's retry policy (number of retries, delays, etc) but it is a global (per-worker) setting https://github.com/golemcloud/golem-wit/blob/main/wit/deps/golem/golem-host.wit#L40-L50

Reply:
Fixed.